packages/openai-compatible/src/openai-compatible-error.ts

```diff
@@ -0,0 +1,19 @@
+import { z } from 'zod';
+import { createJsonErrorResponseHandler } from '@ai-sdk/provider-utils';
+
+// TODO(shaper): Reconcile this with openai-error.ts. We derived from `xai`.
+
+export const openAICompatibleErrorDataSchema = z.object({
+  code: z.string(),
+  error: z.string(),
+});
+
+export type OpenAICompatibleErrorData = z.infer<
+  typeof openAICompatibleErrorDataSchema
+>;
+
+export const openAICompatibleFailedResponseHandler =
+  createJsonErrorResponseHandler({
+    errorSchema: openAICompatibleErrorDataSchema,
+    errorToMessage: data => data.error,
+  });
```

Comment:
make provider-specific. different providers have different structures. have default that matches openai

Reply:
I can see a couple of different ways to do this. The error data schema is further embedded in the schemas used in the chat/completion model classes, so just making the failed response handler configurable isn't sufficient for the goal.

I have in mind to break this PR into two so that we can start landing parts and work more incrementally:
1. initial openai-compatible package work
2. togetherai package atop it

I expect we can land (1) tomorrow after I iterate on your next feedback. We can then put a focused change together to make the error schema/response handler configurable. And we can iterate on (2) until it supports the model/feature combos we feel is good enough for a first ship target.